lib/action_view/component/ar_monkey_patch.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class ActiveRecord::Base
+  def to_component_class
+    class_name = "#{self.class.name}Component"
+
+    class_name.constantize if defined?(class_name)
```

Comment:
ActiveSupport implements [`#safe_constantize`](https://apidock.com/rails/v6.0.0/ActiveSupport/Inflector/safe_constantize) which returns `nil` if the constant isn't defined, making it possible to refactor this method into a one-liner:
```ruby
def to_component_class
  "#{self.class.name}Component".safe_constantize
end
```
This is only a matter of style though, there's nothing wrong with the implementation as it is 🙂